### PR TITLE
Enter type-to-nav for plugin dir

### PR DESCRIPTION
**Pull request by luukvbaal**
RE https://github.com/jarun/nnn/pull/1373#issuecomment-1140461536: I think this covers everything, without too many extra checks. Please check if you agree. I think this is a nice alternative to the `fzplug` plugin.

Also enables auto-enter for `g_state.selbm` and `g_state.runplugin`.

### CODE REVIEW

**Comment by N-R-K** [comment]
Without looking at the code, I'm wondering what usecase does this serve? Having ttn mode on in bookmarks dir allows for quickly navigating the bookmarks. I haven't seen anyone ask for ttn mode on plugins dir.

**Comment by luukvbaal** [comment]
It only enters ttn in plugin selection mode(i.e. pressing <kbd>;Enter</kbd>, not when navigating there menually).

I think it is a fair assumption to make that the user would like to easily filter/select the plugins and ttn provides a native way to do so.

Basically it is a builtin alternative to `fzplug` like I said. Do you think it is undesired behavior?

**Comment by N-R-K** [comment]
> It only enters ttn in plugin selection mode(i.e. pressing ;Enter, not when navigating there menually).
> 
> I think it is a fair assumption to make that the user would like to easily filter/select the plugins and ttn provides a native way to do so.
> 
> Basically it is a builtin alternative to `fzplug` like I said. Do you think it is undesired behavior?

Ah, I misunderstood what this was doing then. And no, I think this is a fair improvement and makes sense.

File: src/nnn.c [hunk 1]
```diff
@@ -6683,6 +6681,12 @@ static bool browse(char *ipath, const char *session, int pkey)
 		setdirwatch();
 	}
 
+	if (g_state.filtermode) {
```
Comment (jarun) [inline src/nnn.c:6684]: A new check every time we jump to begin. This is what I was talking about. I have to think about it.

**Review by jarun — CHANGES_REQUESTED** [review]

**Comment by jarun** [comment]
This adds open condition checks in flows. That's what I was indicating. I'll think about it.

File: src/nnn.c (hunk 1 above)
Comment (luukvbaal) [reply src/nnn.c:6684]: Sure, I understand why you would be apprehensive of adding a check there. I just hoped the convenience factor would outweigh the performance consideration :)

Let me know and I'll raise the PR for the `restorepreview` patch conflict(dependent on this PR) and patch conflict resolve instructions later.

Comment (luukvbaal) [reply src/nnn.c:6684]: Keep in mind that this block will almost never execute, the first check will only evaluate to true when the user presses <kbd>b/;Enter</kbd>.

Comment (jarun) [reply src/nnn.c:6684]: It's just a matter of pressing /.

**Comment by jarun** [comment]
Guys I have decided not to go with this. I will also revert the bm hack. Can't litter the code with roundabout checks to save on a single keypress (`/`).

**Pull request closed without merging**